### ISSUE TITLE
Azure Blob triggered function fails blobContent.trim is not a function for WebApp Application and Server Diagnostic logs

### ISSUE THREAD

**Issue by KyleMartin901** [DataDog/datadog-serverless-functions#110]
**Describe what happened:**
Once a WebApp application or server diagnostic log has been added to the storage account the function errors with Exception: TypeError: blobContent.trim is not a function

```bash
2019-05-01T02:44:48  Welcome, you are now connected to log-streaming service.
2019-05-01T02:45:31.401 [Information] Executing 'Functions.sitecore_cd_server_and_application_logs' (Reason='New blob detected: logs/PROD-CD-APP/2019/05/01/02/efbf18.log', Id=4f01a2c1-a616-4351-8a0b-6806fea15632)
2019-05-01T02:45:32.270 [Error] Executed 'Functions.sitecore_cd_server_and_application_logs' (Failed, Id=4f01a2c1-a616-4351-8a0b-6806fea15632)
Result: Failure
Exception: TypeError: blobContent.trim is not a function
Stack: TypeError: blobContent.trim is not a function
    at module.exports (D:\home\site\wwwroot\sitecore_cd_server_and_application_logs\index.js:44:28)
    at WorkerChannel.invocationRequest (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:16077:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:15967:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19692:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19007:19)
    at D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19106:14
2019-05-01T02:45:32.701 [Information] Executing 'Functions.sitecore_cd_server_and_application_logs' (Reason='New blob detected: logs/PROD-CD-APP/2019/05/01/02/efbf18.log', Id=457e7a33-8df1-464c-9714-46eb843018a4)
2019-05-01T02:45:32.913 [Error] Executed 'Functions.sitecore_cd_server_and_application_logs' (Failed, Id=457e7a33-8df1-464c-9714-46eb843018a4)
Result: Failure
Exception: TypeError: blobContent.trim is not a function
Stack: TypeError: blobContent.trim is not a function
    at module.exports (D:\home\site\wwwroot\sitecore_cd_server_and_application_logs\index.js:44:28)
    at WorkerChannel.invocationRequest (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:16077:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:15967:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19692:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19007:19)
    at D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19106:14
2019-05-01T02:45:33.198 [Information] Executing 'Functions.sitecore_cd_server_and_application_logs' (Reason='New blob detected: logs/PROD-CD-APP/2019/05/01/02/efbf18.log', Id=e9a4ec0f-4d00-4f2a-bf77-57faafbbd5b3)
2019-05-01T02:45:33.571 [Error] Executed 'Functions.sitecore_cd_server_and_application_logs' (Failed, Id=e9a4ec0f-4d00-4f2a-bf77-57faafbbd5b3)
Result: Failure
Exception: TypeError: blobContent.trim is not a function
Stack: TypeError: blobContent.trim is not a function
    at module.exports (D:\home\site\wwwroot\sitecore_cd_server_and_application_logs\index.js:44:28)
    at WorkerChannel.invocationRequest (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:16077:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:15967:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19692:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19007:19)
    at D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19106:14
2019-05-01T02:45:34.633 [Information] Executing 'Functions.sitecore_cd_server_and_application_logs' (Reason='This function was programmatically called via the host APIs.', Id=8928a007-3185-43d0-917b-fa97561b3274)
2019-05-01T02:45:34.900 [Error] Executed 'Functions.sitecore_cd_server_and_application_logs' (Failed, Id=8928a007-3185-43d0-917b-fa97561b3274)
Result: Failure
Exception: TypeError: blobContent.trim is not a function
Stack: TypeError: blobContent.trim is not a function
    at module.exports (D:\home\site\wwwroot\sitecore_cd_server_and_application_logs\index.js:44:28)
    at WorkerChannel.invocationRequest (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:16077:26)
    at ClientDuplexStream.WorkerChannel.eventStream.on (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:15967:30)
    at ClientDuplexStream.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at ClientDuplexStream.Readable.push (_stream_readable.js:219:10)
    at Object.onReceiveMessage (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19692:19)
    at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19007:19)
    at D:\Program Files (x86)\SiteExtensions\Functions\2.0.12427\32bit\workers\node\worker-bundle.js:19106:14
2019-05-01T02:46:48  No new trace in the past 1 min(s).
```

**Describe what you expected:**
That the newly detected file would be read and data sent to my Datadog account.

**Steps to reproduce the issue:**
Enable WebApp Diagnostic Logs to blob store. Create Datadog function app with blob trigger.

**Comment by fogers777** [comment]
@KyleMartin901 i had same issue, fix was to replace var logs = blobContent.trim().split('\n'); with var logs = blobContent.**toString()**.trim().split('\n');

**Comment by NilsBunge** [comment]
Hello @KyleMartin901, 

First of all thanks for reporting this issue. 
The test we made were indeed with string logs so we missed that use case. Thanks a lot @dopsy for the suggestion, I'll make sure we merge it into the main function.

**Comment by KyleMartin901** [comment]
Thanks @dopsy yep adding tostring fixed my issue. 

@NBParis no worries will be good to see it merged in. 

Thanks for reaching both of you was great to get it up and running now to automate the deployment with our web apps 😬

**Comment by NilsBunge** [comment]
Hello, 

The PR has been merged and the function should now properly handle blob that are not strings.
(Note that the code we made is slightly different from a `toString()` that could have some issues in specific cases)
I'll go ahead and close this issue but feel free to reach out either with a new issue or to support@datadoghq.com if you face any other problem.

Once again, thanks a lot for reporting the issue and suggesting a fix.
